src/XMakeBuildEngine/BackEnd/Node/OutOfProcNode.cs

```diff
@@ -234,15 +234,28 @@ internal static bool IsOutOfProcNode
 
         /// <summary>
         /// Starts up the node and processes messages until the node is requested to shut down.
+        /// Assumes no node reuse.
         /// </summary>
         /// <param name="shutdownException">The exception which caused shutdown, if any.</param>
         /// <returns>The reason for shutting down.</returns>
         public NodeEngineShutdownReason Run(out Exception shutdownException)
         {
+            return Run(false, out shutdownException);
+        }
+
+
+        /// <summary>
+        /// Starts up the node and processes messages until the node is requested to shut down.
+        /// </summary>
+        /// <param name="enableReuse">Whether this node is eligible for reuse later.</param>
+        /// <param name="shutdownException">The exception which caused shutdown, if any.</param>
+        /// <returns>The reason for shutting down.</returns>
+        public NodeEngineShutdownReason Run(bool enableReuse, out Exception shutdownException)
```

Comment:
Doesn't anyone have to call this method with enableReuse set to true, when node reuse is actually desired? With the current code it looks like nodeReuse is always set to false.